### PR TITLE
shift: add page

### PR DESCRIPTION
**Pull request by stku1985**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

**Comment by bl-ue** [comment]
@stku1985 we only use single-line examples, enclosed in single back ticks (\`...\` rather than \`\`\`...\`\`\`)

File: pages/common/shift.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+> See <https://www.gnu.org/software/bash/manual/bash.html> for more information.
+
+- Shift arguments by one place; argument `$1` is dropped, argument `$2` is now `$1`:
+
+```bash
+# prints two times the same output
+echo $2
+shift
+echo $1 
+```
```
Comment (bl-ue) [inline pages/common/shift.md:13]: @stku1985 I'd do something like this:
```suggestion
- Move all arguments back by one (dropping the first argument):

`shift`
```

**Comment by stku1985** [comment]
@bl-ue thanks for the quick reply. I understand that single line code examples are a community standard; however since `shift` is strictly for scripting, the examples need some context imo.

What is your suggestion?

File: pages/common/shift.md (hunk 1 above)
Comment (stku1985) [reply pages/common/shift.md:13]: Ok I will do so. I think this reduces the intelligibility of the page, but I also understand that you want to keep consistency between pages.

**Comment by bl-ue** [comment]
@stku1985 if you can, please avoid force-pushing.  It's not necessary as we automatically squash commits before merging and it can make reviewing a bit difficult 🙂

File: pages/common/shift.md (hunk 1 above)
Comment (bl-ue) [reply pages/common/shift.md]: Why does it decrease the intelligibility @stku1985?

**Comment by stku1985** [comment]
@bl-ue I will keep that in mind for the next time. Sorry for the confusion.

File: pages/common/shift.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# shift
+
+> A bash built-in command that shifts the (numbered) arguments of a function or script by a specified number of places.
```
Comment (bl-ue) [inline pages/common/shift.md:3]:
```suggestion
> Bash built-in command that shifts the arguments passed to the calling function or script by a specified number of places.
```

File: pages/common/shift.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+> See <https://www.gnu.org/software/bash/manual/bash.html> for more information.
+
+- Move arguments by one place dropping the first argument:
+
+`shift`
+
+- Move arguments by {{n}} places dropping the first {{n}} arguments:
+
+`shift {{n}}`
```
Comment (bl-ue) [inline pages/common/shift.md:12]: We don't typically use token syntax inside example descriptions.  I'd do this, in this case:
```suggestion
- Move arguments by N places dropping the first N arguments:

`shift {{N}}`
```

File: pages/common/shift.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# shift
+
+> A bash built-in command that shifts the (numbered) arguments of a function or script by a specified number of places.
+> See <https://www.gnu.org/software/bash/manual/bash.html> for more information.
```
Comment (bl-ue) [inline pages/common/shift.md:4]: We have a standard syntax for more information links:
```suggestion
> More information: <https://www.gnu.org/software/bash/manual/bash.html>.
```

**Review by bl-ue — CHANGES_REQUESTED** [review]
Thanks for the new page @stku1985!  I'm really surprised this one wasn't already added 😄

I've left a few comments below.

File: pages/common/shift.md (hunk 1 above)
Comment (stku1985) [reply pages/common/shift.md]: Well I think scripting commands (such as shift, return, continue, for, ...) need some context. For experienced bash users this short example is fine; for people new to bash scripting my original version or even a whole function example would be easier to grasp. 

But don't worry; either way is fine from my point of view.

Comment (bl-ue) [reply pages/common/shift.md]: I agree so much with that—but that's not what tldr-pages was originally invented for.  It's really for documented syntax of a command, not really how to use it in conjunction with other commands.

**Review by bl-ue — APPROVED** [review]
Lovely and much needed! 😋

File: pages/common/shift.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# shift
+
+> Bash built-in command that shifts the arguments passed to the calling function or script by a specified number of places.
+> More information: <https://www.gnu.org/software/bash/manual/bash.html>.
```
Comment (marchersimon) [inline pages/common/shift.md:4]:
```suggestion
> More information: <https://www.gnu.org/software/bash/manual/bash.html#index-shift>.
```

Maybe add an anchor to the link, so the command can be found more easily.

**Review by marchersimon — APPROVED** [review]

**Pull request merged**